tardis/transport/montecarlo/packet_trackers.py

```diff
@@ -156,3 +156,50 @@ def rpacket_trackers_to_dataframe(rpacket_trackers):
         index=pd.MultiIndex.from_arrays(index_array, names=["index", "step"]),
         columns=df_dtypes.names,
     )
+
+
+rpacket_last_interaction_tracker_spec = [
+    ("index", int64),
+    ("r", float64),
+    ("nu", float64),
+    ("energy", float64),
+    ("shell_id", int64),
+    ("interaction_type", int64),
+]
+
+
+@jitclass(rpacket_last_interaction_tracker_spec)
+class RPacketLastInteractionTracker(object):
```

Comment:
I'm not sure If I'm happy with adding a separate class for Something that really is only an edge case of the RPacketTracker class, i.e with length 1. I haven't looked at it in detail, but I'm sure this should be possible by adding a bit of additional logic to the RPacketTracker, e.g. by preventing the resizing of the list and always only writing to the last (and only element). A nice side effect could be that one could add the Option to use this to track the last n packets. Also this way we won't have to duplicate the auxiliary functions, e.g. i/o.

Ultimately I'm just concerned that this duplication will lead to complications down the line. In case there are specific reasons for doing it this way, I would at least add a comment explaining the reasoning behind this approach.

Reply:
I agree that the LastInteraction is just a particular case of all the interactions, but this is the interaction info that the sdec_plot uses to determine line elements. Also, the currently worked-on visualization tool is based on the last interaction. So, creating a specific class for the last interaction is justified in this sense, though I am still new to this field, and my reasoning might be flawed. Also, if, in the future, we want to add more functions specific to the last interaction, it can be quickly done. Since RPacketTracker and RPacketInteractionTracker are independent, the last interaction class can be easily removed if everyone agrees. This is just abstract reasoning, and @wkerzendorf might provide more concrete reasoning for why this is needed.